connectors-home/modules/ROOT/pages/introduction/intro-config-oauth2.adoc

```diff
@@ -0,0 +1,189 @@
++
+* The *Authorization URL* must be reachable from the user's browser.
+* The provided scopes for *Scopes* are supported.
+* The *Resource Owner ID* (if provided) must be registered with the OAuth provider.
+* The full *Callback Path* must be registered with the OAuth provider.
+. Enter your username, password, and other credentials that the OAuth provider accepts, such as 2FA requests. If the authentication is successful, the OAuth provider might ask permission for the application to access your account. This typically occurs the first time you authenticate.
+. Once the OAuth provider authenticates the user and authorizes access to the connector, the user's web browser is redirected to the callback URL that is registered with the OAuth provider, for example, `\http://localhost:8080/callback`.
++
+Ensure that the user running the flow has access to that server and endpoint, otherwise the authorization approval does not reach the connector.
++
+You can use `localhost` for testing locally, but you cannot use `localhost` for production environments and cloud deployment, for example, if your connector application is running on `\http://my.server.com`, you must specify the *External Callback Url* field with the `\http://my.server.com/callback` value. The `\http://my.server.com/callback` URL must also be registered with the OAuth provider.
+. Once the user's browser successfully reaches the callback URL, the connector obtains an authorization code that is automatically exchanged for an access token using the URL configured in the *Access Token URL* field. This token is long-lived and is automatically refreshed by the connector indefinitely (if the OAuth provider allows so).
+
+The connector is now fully authorized and can use any service that accepts its new access token issued by the OAuth provider.
+
+== Example: How to Configure OAuth 2.0 For Gmail Connector
+
+The following example shows how to configure OAuth 2.0 for Gmail Connector.
+
+=== Configuring the OAuth Provider
+
+To configure the OAuth provider for Gmail Connector, you can use Google Cloud Console. The following steps show how to configure the OAuth provider:
+
+NOTE: If you haven't enabled the Gmail API previously, you must create a new project and enable the Gmail API in the *Enabled APIs & Services* menu section.
+
+. Add a new client (your connector application) to Google OAuth 2.0. To do so, navigate to *APIs & Services > Credentials > Create Credentials > OAuth Client ID*.
+. Create a new *Web Application*.
+. Enter a name, for example, `OAuthDemoApp`.
+. Add the following URL to the *Authorized redirect URIs*: `\http://127.0.0.1:8080/callback`.
+. Click *Create* and copy the new client ID and client secret for later use.
+. Click *OK*.
+
+=== Configuring the Connector
+
+The following steps show how to configure the connector with OAuth 2.0:
```

Comment:
This example is for configuring the connector using XML. Shouldn't we also show how to do this in the GUI?